chapter8.adoc

```diff
@@ -1,27 +1,66 @@
 [[chapter8]]
 [[Smsdedbg]]
-== `Smsdedbg`: Supervisor Domain External Debug
+== Supervisor Domain External Trace and Debug
 
-The <<MSDCFG>> CSR contains the `sdedbgalw` bit that controls whether the
-current scheduled SD is allowed to be external-debugged cite:[ExtDbg]. This bit
-is context switched (along with rest of the `msdcfg`) per SD.
+The <<MSDCFG>> CSR contains the `sdedbgalw` and `sdetrcalw` bits that control
+whether the current scheduled SD is allowed to be external-debugged or to be
+external-traced respectively.
 
-=== `Smsdedbg` interaction with external debug security controls (Informative)
+=== `Smsdedbg`: External Debug allowed control for Supervisor Domain
+
+When M-mode external debug is enabled, all supervisor domains may also be
+debugged by an external debugger irrespective of the configuration held in
+`msdcfg.SDEDBGALW`.
+
+When M-mode external debug is disabled, whether execution at privilege modes
+less than `M-mode` may be debugged by an external debugger depends on the
+configuration held in `msdcfg.SDEDBGALW`.
+
+When `msdcfg.SDEDBGALW` = 0, external debug is disallowed for the SD. Abstract
+commands and halt request from the debug module are suppressed and stay pending
+while the SD is active.
+
+When `msdcfg.SDEDBGALW` = 1 then external debug of privilege modes less than
+`M-mode` is allowed for the SD, and:
+
+* A halt request may transition the hart to Debug Mode.
+* Abstract commands and program buffer execution can access state of privilege
+modes less than `M-mode`.
+* Read and Write of `Sdtrig` and `Sdext` CSRs are allowed.
+* Debugger memory accesses occur with either `S-mode` or `U-mode` privilege (as
+if `aamvirtual` = 1 and `MPP` != `M-mode`).
```

Comment:
The problem brought up here is not yet addressed. It will break debug use cases.

Reply:
The key word is "as if". The "Access Memory" command is specified in section 3.7.1.3. This command lets debugger perform memory access. The arg0 of this command has aamvirtual which when set to 1 indicates the address is a virtual address - and these are translated the same way as they would be from M-mode, with MPRV set. The abstract commands can only execute at U or S mode when medbgen is 0 and they will always execute as-if aamvirtual = 1 and MPP !=M-mode. Please clarify what debug use cases get affected.

Reply:
The statement says "S-mode or U-mode", but basically which one? According to value in MPP or what else? The spec needs clear definition on access privilege. Otherwise, it leaves impression that use cases are affected.

Reply:
This point must be clearly defined in main content. But currently not a clue.

Reply:
please clarify what is unclear in the statement in the text:
Debugger memory accesses occur with either `S-mode` or `U-mode` privilege (as if `aamvirtual` = 1 and `MPP` != `M-mode`).